include/zephyr/cache.h

```diff
@@ -341,8 +257,10 @@ static inline int sys_cache_instr_flush_range(void *addr, size_t size)
  * @retval -ENOTSUP If not supported.
  * @retval -errno Negative errno for other failures.
  */
-__syscall int sys_cache_data_invd_range(void *addr, size_t size);
-static inline int z_impl_sys_cache_data_invd_range(void *addr, size_t size)
+__syscall int sys_cache_data_invd_range(void *addr, size_t size)
+	__attribute__((always_inline));
```

Comment:
Should use `ALWAYS_INLINE` for portability.

Reply:
Yes, but that would confuse `scripts/build/parse_syscalls.py` that would be unable to parse anything (this is why the attribute is also put at the end). So, either we live with this or we have to modify the parsing script to account for `ALWAYS_INLINE`.

Reply:
Sounds like a `parse_syscalls.py` issue that should be resolved.

Reply:
done, but be aware that my python-foo is not strong :D